source/robot_model/include/robot_model/Model.hpp

```diff
@@ -122,19 +122,23 @@ class Model {
    * @brief Compute the forward kinematics, i.e. the pose of certain frames from the joint positions
    * @param joint_positions the joint state of the robot
    * @param frame_ids ids of the frames at which to extract the pose
+   * @param ref_frame_id id of the reference frame
    * @return the desired poses
    */
   std::vector<state_representation::CartesianPose> forward_kinematics(const state_representation::JointPositions& joint_positions,
-                                                                      const std::vector<unsigned int>& frame_ids);
+                                                                      const std::vector<unsigned int>& frame_ids,
+                                                                      const unsigned int ref_frame_id = 0);
```

Comment:
I'm temped to say that the reference frame has to be a vector too @eeberhard what do you think?
Or should we provide a vector of pairs?

Reply:
Do you mean like multiple reference frames so we get back multiple viewpoints of the same frame, or rather we will get back num_of_frame_ids * num_of_ref_frame_ids back

Reply:
IMO a common reference frame is sufficient. From an API side, it's simpler this way. I don't know how common it would be to request each frame in a separate reference frame, and even then you can just use state representation after the fact to transform a frame secondarily if it's super necessary (i.e., get elbow and wrist in base frame, then find wrist in elbow frame afterwards)